src/main/java/org/prebid/server/floors/BasicPriceFloorResolver.java

```diff
@@ -369,17 +372,35 @@ private static List<String> gptAdSlotFromImp(Imp imp) {
         return Collections.singletonList(gptAdSlot);
     }
 
-    private static List<String> pbAdSlotFromImp(Imp imp) {
+    private static List<String> adUnitCodeFromImp(Imp imp) {
         final ObjectNode impExt = imp.getExt();
-
+        final String tagId = imp.getTagid();
         if (impExt == null) {
-            return null;
+            return Collections.singletonList(tagId);
```

Comment:
here `tagId` can be null, so you will return `List.of(null)`, i think you can add null check for `impExt` to `stringByPath`, remove this `if` and just return `Collections.emptyList()` at the end.